rotkehlchen/data/importer.py

```diff
@@ -463,7 +465,298 @@ def import_cryptocom_csv(self, filepath: Path) -> Tuple[bool, str]:
+                identifier=0,  # whatever is not used at insertion
+                timestamp=timestamp,
+                action_type=LedgerActionType.INCOME,
+                location=Location.BLOCKFI,
+                amount=amount,
+                asset=asset,
+                rate=None,
+                rate_asset=None,
+                link=None,
+                notes=f'{entry_type} from BlockFi',
+            )
+            self.db_ledger.add_ledger_action(action)
+        elif entry_type == 'Trade':
+            pass
+        else:
+            raise UnsupportedCSVEntry(f'Unsuported entry {entry_type}. Data: {csv_row}')
+
+    def import_blockfi_transactions_csv(self, filepath: Path) -> Tuple[bool, str]:
+        """
+        Information for the values that the columns can have has been obtained from
+        https://github.com/BittyTax/BittyTax/blob/06794f51223398759852d6853bc7112ffb96129a/bittytax/conv/parsers/blockfi.py#L67
+        """
+        with open(filepath, 'r', encoding='utf-8-sig') as csvfile:
+            data = csv.DictReader(csvfile)
+            for row in data:
+                try:
+                    self._consume_blockfi_entry(row)
+                except UnknownAsset as e:
+                    self.db.msg_aggregator.add_warning(
+                        f'During BlockFi CSV import found action with unknown '
+                        f'asset {e.asset_name}. Ignoring entry',
+                    )
+                    continue
+                except DeserializationError as e:
+                    self.db.msg_aggregator.add_warning(
+                        f'Deserialization error during BlockFi CSV import. '
+                        f'{str(e)}. Ignoring entry',
+                    )
+                    continue
+                except UnsupportedCSVEntry as e:
+                    self.db.msg_aggregator.add_warning(str(e))
+                    continue
+                except KeyError as e:
+                    return False, str(e)
+        return True, ''
+
+    def _consume_blockfi_trade(self, csv_row: Dict[str, Any]) -> None:
+        """
+        Consume the file containing only trades from BlockFi. As per my investigations
+        (@yabirgb) this file can only contain confirmed trades.
+        - UnknownAsset
+        - DeserializationError
+        """
+        timestamp = deserialize_timestamp_from_date(
+            date=csv_row['Date'],
+            formatstr='%Y-%m-%d %H:%M:%S',
+            location='BlockFi',
+        )
+
+        buy_asset = symbol_to_asset_or_token(csv_row['Buy Currency'])
+        buy_amount = deserialize_asset_amount(csv_row['Buy Quantity'])
+        sold_asset = symbol_to_asset_or_token(csv_row['Sold Currency'])
+        sold_amount = deserialize_asset_amount(csv_row['Sold Quantity'])
+        if sold_amount == ZERO:
+            log.debug(f'Ignoring BlockFi trade with sold_amount equal to zero. {csv_row}')
+            return
+        rate = Price(buy_amount / sold_amount)
```

Comment:
You could have also made a rate = 0 to not lose data here, but it's a small nitpick. Either way useless entry.